Micromet Network
# Match Microbe Genes and Microbe Reactions

&rarr; Goal: match microbes to reactions via the genes involved

In [1]:
import pandas as pd

In [2]:
gene_reactions_file_path = "cleaned_tables/gene_reactions_cleaned.csv"
microbe_genes_file_path = "/Users/vivy/PycharmProjects/NutritionDiseaseKnowledgeGraph/Databases/VMH/Genes/vmh_genes_from_microbe_all.csv"

In [3]:
gene_reactions = pd.read_csv(gene_reactions_file_path)
microbe_genes = pd.read_csv(microbe_genes_file_path)

In [4]:
gene_reactions

Unnamed: 0 reaction_abbreviation        gene_number
0                 0             13DAMPPOX             8639.1
1                 1             42A12BOOX             8639.1
2                 2              NMPTRCOX             8639.1
3                 3             13DAMPPOX               26.1
4                 4             42A12BOOX               26.1
...             ...                   ...                ...
2133292     2133292                 GALt4   521000.6.peg.536
2133293     2133293                 GALt4  500637.6.peg.1428
2133294     2133294                 GALt4  500637.6.peg.4927
2133295     2133295                 GALt4  471874.6.peg.2439
2133296     2133296                 GALt4  762983.3.peg.1596

[2133297 rows x 3 columns]

In [5]:
microbe_genes

Unnamed: 0         gene_number                            microbe
0                1    592010.4.peg.619   Abiotrophia_defectiva_ATCC_49176
1                2     592010.4.peg.75   Abiotrophia_defectiva_ATCC_49176
2                3    592010.4.peg.380   Abiotrophia_defectiva_ATCC_49176
3                4   592010.4.peg.2638   Abiotrophia_defectiva_ATCC_49176
4                5   592010.4.peg.1101   Abiotrophia_defectiva_ATCC_49176
...            ...                 ...                                ...
639136          33  1002368.3.peg.1541  Yokenella_regensburgei_ATCC_43003
639137          34  1002368.3.peg.1542  Yokenella_regensburgei_ATCC_43003
639138          35  1002368.3.peg.1543  Yokenella_regensburgei_ATCC_43003
639139          36  1002368.3.peg.1435  Yokenella_regensburgei_ATCC_43003
639140          37    g.23117.CDS.2157  Yokenella_regensburgei_ATCC_43003

[639141 rows x 3 columns]

In [6]:
df_merge = pd.merge(gene_reactions, microbe_genes, how="inner", on="gene_number")
print("Number of matched microbe reactions: " + str(df_merge.gene_number.count()))

Number of matched microbe reactions: 2570205


In [7]:
unique_genes_microbes = pd.DataFrame(microbe_genes.gene_number.unique())
unique_genes_microbes.columns = ['gene_number_microbes']

unique_genes_reactions = pd.DataFrame(gene_reactions.gene_number.unique())
unique_genes_reactions.columns = ['gene_number_reactions']

In [8]:
df_merge = pd.merge(unique_genes_microbes, unique_genes_reactions, how="outer", left_on='gene_number_microbes', right_on='gene_number_reactions')

In [9]:
len(df_merge[df_merge.gene_number_reactions.isnull()])

0

**&rarr; All microbe genes could be matched to a reaction**

In [10]:
len(df_merge[df_merge.gene_number_microbes.isnull()])

3693

In [11]:
df_merge[df_merge.gene_number_microbes.isnull()]

gene_number_microbes gene_number_reactions
638293                  NaN                8639.1
638294                  NaN                  26.1
638295                  NaN                 314.2
638296                  NaN                 314.1
638297                  NaN                1591.1
...                     ...                   ...
641981                  NaN                5973.1
641982                  NaN                1635.1
641983                  NaN              286297.1
641984                  NaN               10560.1
641985                  NaN            gap_filled

[3693 rows x 2 columns]

In [12]:
microbe_genes[microbe_genes.gene_number == '8639.1']

Empty DataFrame
Columns: [Unnamed: 0, gene_number, microbe]
Index: []

**&rarr; About 3693 reactions are not catalyzed by a microbe in this database as we can't find a corresponding microbe gene. It seems as if the genes from reactions that could not be matched are human (or from another animal)**